<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Copy Number Event Detection with CNprep • CNprep</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Copy Number Event Detection with CNprep">
<meta property="og:description" content="CNprep">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CNprep</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">2.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/CNprep.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/KrasnitzLab/CNprep/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="CNprep_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Copy Number Event Detection with CNprep</h1>
                        <h4 data-toc-skip class="author">Guoli Sun, Pascal Belleau, Astrid Deschênes et Alex Krasnitz</h4>
            
            <h4 data-toc-skip class="date">2021-06-22</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/KrasnitzLab/CNprep/blob/master/vignettes/CNprep.Rmd"><code>vignettes/CNprep.Rmd</code></a></small>
      <div class="hidden name"><code>CNprep.Rmd</code></div>

    </div>

    
    
<p><br><strong>Package</strong>: CNprep<br><strong>Authors</strong>: Alex Krasnitz [aut], Pascal Belleau [aut, cre] (<a href="https://orcid.org/0000-0002-0802-1071" class="uri">https://orcid.org/0000-0002-0802-1071</a>), Astrid Deschênes [aut] (<a href="https://orcid.org/0000-0001-7846-6749" class="uri">https://orcid.org/0000-0001-7846-6749</a>), Guoli Sun [aut]<br><strong>Version</strong>: 2.1.1<br><strong>Compiled date</strong>: 2021-06-22<br><strong>License</strong>: GPL-2<br></p>
<div id="license" class="section level1">
<h1 class="hasAnchor">
<a href="#license" class="anchor"></a>License</h1>
<p>The <em>CNprep</em> package and the underlying <em>CNprep</em> code are distributed under the <a href="https://www.gnu.org/licenses/old-licenses/gpl-2.0.txt">GPL-2 License</a>. You are free to use and redistribute this software.</p>
</div>
<div id="citing" class="section level1">
<h1 class="hasAnchor">
<a href="#citing" class="anchor"></a>Citing</h1>
<p>If you use this package for a publication, we ask you to cite the following:</p>
<blockquote>
<p>Belleau P, Deschênes A, Sun G et al. CNprep: Copy number event detection [version 1; not peer reviewed]. F1000Research 2020, 9:743 (poster) (<a href="https://doi.org/10.7490/f1000research.1118065.1" class="uri">https://doi.org/10.7490/f1000research.1118065.1</a>)</p>
</blockquote>
</div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>The <em>CNprep</em> package has been implemented to identify, through a multi-step process, segments with significant copy number deviation from normal state. To this end, <em>CNprep</em> uses both DNA copy number initial form (copy number as a noisy function of genomic position, also known as bins) and their approximation by a piecewise-constant function (segmentation).</p>
<p>Copy number variations (CNVs) are associated with a wide spectrum of pathological conditions and complex traits, such as Tourette Disorder <span class="citation">(Wang et al. 2018)</span> and especially cancer <span class="citation">(Pelttari et al. 2018; Franch-Expósito et al. 2018)</span>. Furthermore, CNVs can potentially be used as biomarkers for cancer diagnosis <span class="citation">(Pan et al. 2018)</span>.</p>
<p>Since late 1990s, one commonly used technology to detect CNVs is the Comparative Genomic Hybridization (CGH) microarray <span class="citation">(Carter 2007)</span>. From those microarrays, software such as <em>DNACopy</em> <span class="citation">(Seshan and Olshen 2019)</span> can identify regions of gained and lost copy number. However, because of the low resolution of the array, detection of short CNVs is limited <span class="citation">(Duan et al. 2013)</span>. The availability and low cost of Next-Generation Sequencing (NGS) have enabled precise detection of CNVs for the entire genome with the exclusion of few hard to map regions. This lead to the development of software adapted to whole-genome and whole-exome sequencing, such as <em>CNVkit</em> <span class="citation">(Talevich et al. 2016)</span> and <em>exomecopy</em> <span class="citation">(Love 2019)</span>.</p>
<p>In the analysis of bulk tumor DNA cells, not all cells share the same ploidy or harbor the same copy number alterations. Hence, segments identified may not share integer copy number. So, evaluating the value of the copy number corresponding to the normal state of the cells, relative to which somatic alterations occur, is difficult <span class="citation">(Kendall and Krasnitz 2014)</span>.</p>
<p>The <em>CNprep</em> package has been implemented to identify,</p>
<p>TODO</p>
</div>
<div id="installing-and-loading-cnprep-package" class="section level1">
<h1 class="hasAnchor">
<a href="#installing-and-loading-cnprep-package" class="anchor"></a>Installing and loading <em>CNprep</em> package</h1>
<p>The <em>CNprep</em> package can be installed from github website using <em>devtools</em> package:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/">devtools</a></span><span class="op">)</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"belleau/CNprep"</span><span class="op">)</span></code></pre></div>
<p>As with any R package, the <em>CNprep</em> package has to be loaded with the following command:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/KrasnitzLab/CNprep">CNprep</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="processing-copy-number" class="section level1">
<h1 class="hasAnchor">
<a href="#processing-copy-number" class="anchor"></a>Processing copy number</h1>
<div id="processing-copy-number-for-bulk-tumor" class="section level2">
<h2 class="hasAnchor">
<a href="#processing-copy-number-for-bulk-tumor" class="anchor"></a>Processing copy number for bulk tumor</h2>
</div>
<div id="processing-copy-number-for-bulk-tumor-when-cancer-free-tissues-available" class="section level2">
<h2 class="hasAnchor">
<a href="#processing-copy-number-for-bulk-tumor-when-cancer-free-tissues-available" class="anchor"></a>Processing copy number for bulk tumor when cancer-free tissues available</h2>
</div>
</div>
<div id="create-and-apply-a-mask-to-copy-number-events" class="section level1">
<h1 class="hasAnchor">
<a href="#create-and-apply-a-mask-to-copy-number-events" class="anchor"></a>Create and apply a mask to copy number events</h1>
<p>In the <em>CNprep</em> package, the <em>makeCNPmask()</em> and <em>applyCNPmask()</em> functions are complementary methods.</p>
<p>The <em>makeCNPmask()</em> enables the identification of genomic regions of interest. Those regions can be of interest because those are regions are not specific to the analyzed condition(s), as an example, common copy number alterations that are not related to cancer should be removed when alterations happening in cancer tissues is of interest. Those regions can also be of interest because of their high recurrence in a specific cohort.</p>
<p>The <em>applyCNPmask()</em> function removes the regions identified by the <em>makeCNPmask()</em> function from the dataset.</p>
<div id="remove-non-somatic-copy-number-events-when-matching-normal-profile-is-available" class="section level2">
<h2 class="hasAnchor">
<a href="#remove-non-somatic-copy-number-events-when-matching-normal-profile-is-available" class="anchor"></a>Remove non-somatic copy number events when matching normal profile is available</h2>
<p>Copy number alterations are not always specific to tumor tissues. Some alterations can also be present in the cancer-free tissues, those are called inherited alterations. Yet, somatic copy number alterations, specific to cancer, are often the events of interest.</p>
<p>In the <em>CNprep</em> package, the <em>makeCNPmask()</em> function has been implemented to identify all inherited alterations. This function requires the copy number profile of a matching normal tissue.</p>
<p>TODO explain how to</p>
</div>
<div id="remove-non-somatic-copy-number-events-when-matching-normal-profile-is-not-available-but-multiple-cancer-free-profiles-are-available" class="section level2">
<h2 class="hasAnchor">
<a href="#remove-non-somatic-copy-number-events-when-matching-normal-profile-is-not-available-but-multiple-cancer-free-profiles-are-available" class="anchor"></a>Remove non-somatic copy number events when matching normal profile is not available but multiple cancer-free profiles are available</h2>
<p>When matching cancer-free profile is not available, the identification of inherited alterations is based on the population-wide inherited copy number data. To be able to do so, a cohort of cancer-free profiles is needed.</p>
<p>TODO explain how to</p>
</div>
<div id="identify-recurrent-copy-number-events-in-a-cohort" class="section level2">
<h2 class="hasAnchor">
<a href="#identify-recurrent-copy-number-events-in-a-cohort" class="anchor"></a>Identify recurrent copy number events in a cohort</h2>
<p>When multiple bulk cancer tissues DNA profiles are available, the <em>makeCNPmask()</em> function can be used to identify the recurrent copy number.</p>
<p>The <em>makeCNPmask()</em> function takes as inputs:</p>
<ol style="list-style-type: decimal">
<li>a table of copy number events with genomic positions for all the profiles</li>
<li>an upper threshold</li>
<li>a lower threshold</li>
</ol>
<p>As a first step, interval regions where the event count is above the lower threshold are identified. Among those retained regions, the function only returns the interval regions that contain at least one sub-interval with the count above the upper threshold.</p>
<p>Hence, the final mask is a set of genomic intervals that passed both thresholds.</p>
<p>As an example, a mask in created from copy number events detected in 10 cancer samples. In Figure @ref(fig:demoMask), the top section represent all copy number events detected in the 10 samples for a small region of chromosome 1. The middle section shows the frequency of the events with the lower and upper thresholds selected for this analysis. The bottom section displays the result of the <em>makeCNPmask()</em> function for this dataset when the lower threshold is set to 0.25 and upper threshold to 0.45.</p>
<div class="figure" style="text-align: center">
<img src="figures/DemoMask01.jpeg" alt="Example of makeCNPmask() function. Copy number events detected in 10 samples in top section. Frequency of the events with the selected lower and upper thresholds in the middle section. Result of the makeCNPmask() function for those thresholds in the bottom section." width="420"><p class="caption">
Example of makeCNPmask() function. Copy number events detected in 10 samples in top section. Frequency of the events with the selected lower and upper thresholds in the middle section. Result of the makeCNPmask() function for those thresholds in the bottom section.
</p>
</div>
<p>Using an upper threshold of 0.55 on the same copy number events results in only one genomic region in the mask, as shown in Figure @ref(fig:demoMask2). This region is the only one to respect both lower and upper thresholds.</p>
<div class="figure" style="text-align: center">
<img src="figures/DemoMask02.jpeg" alt="Example of makeCNPmask() function with stringent parameters. Copy number events detected in 10 samples in top section. Frequency of the events with the selected lower and upper thresholds in the middle section. Result of the makeCNPmask() function for those thresholds in the bottom section." width="420"><p class="caption">
Example of makeCNPmask() function with stringent parameters. Copy number events detected in 10 samples in top section. Frequency of the events with the selected lower and upper thresholds in the middle section. Result of the makeCNPmask() function for those thresholds in the bottom section.
</p>
</div>
<p>The following section demonstrates the usage of <em>makeCNPmask()</em> on a more complex dataset composed of 1203 breast cancer samples.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Needed libraries</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">GenomicRanges</span><span class="op">)</span>
<span class="co">## Loading required package: IRanges</span>
<span class="co">## </span>
<span class="co">## Attaching package: 'IRanges'</span>
<span class="co">## The following object is masked from 'package:grDevices':</span>
<span class="co">## </span>
<span class="co">##     windows</span>
<span class="co">## Loading required package: GenomeInfoDb</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/gtrellis">gtrellis</a></span><span class="op">)</span>
<span class="co">## Loading required package: grid</span>
<span class="co">## ========================================</span>
<span class="co">## gtrellis version 1.23.1</span>
<span class="co">## Bioconductor page: http://bioconductor.org/packages/gtrellis/</span>
<span class="co">## Github page: https://github.com/jokergoo/gtrellis</span>
<span class="co">## Documentation: http://bioconductor.org/packages/gtrellis/</span>
<span class="co">## </span>
<span class="co">## If you use it in published research, please cite:</span>
<span class="co">## Gu, Z. gtrellis: an R/Bioconductor package for making genome-level </span>
<span class="co">##   Trellis graphics. BMC Bioinformatics 2016.</span>
<span class="co">## </span>
<span class="co">## This message can be suppressed by:</span>
<span class="co">##   suppressPackageStartupMessages(library(gtrellis))</span>
<span class="co">## ========================================</span>

<span class="co">## Load example containing the copy number events for 1203 samples</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">cnpexample</span><span class="op">)</span>

<span class="co">## Retaining only the amplification events</span>
<span class="va">ampCNP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">cnpexample</span>, <span class="va">copy.num</span> <span class="op">==</span> <span class="st">"amp"</span><span class="op">)</span>

<span class="co">## The amplification table has 4 columns:</span>
<span class="co">##     copy.num identifies the type of event</span>
<span class="co">##     chrom is the number identifying the chromosome</span>
<span class="co">##     chrom.start is the starting position of the event</span>
<span class="co">##     chrom.end is the ending position of the event</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">ampCNP</span>, n<span class="op">=</span><span class="fl">3</span><span class="op">)</span>
<span class="co">##    copy.num chrom chrom.start chrom.end</span>
<span class="co">## 2       amp     1    12842827  13439275</span>
<span class="co">## 5       amp     2    86198238  86575258</span>
<span class="co">## 16      amp     5      772772    860239</span>

<span class="co">## Create a mask using the amplification events.</span>
<span class="co">## The name of the columns for the chromosome identification </span>
<span class="co">## (chromCol parameter), the starting and ending positions </span>
<span class="co">## (startCol and endCol parameters) must be given.</span>
<span class="co">## The number of samples (nProf parameter) is also needed.</span>
<span class="va">ampCNPmask</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeCNPmask.html">makeCNPmask</a></span><span class="op">(</span>imat<span class="op">=</span><span class="va">ampCNP</span>, chromCol<span class="op">=</span><span class="st">"chrom"</span>, 
    startCol<span class="op">=</span><span class="st">"chrom.start"</span>, endCol<span class="op">=</span><span class="st">"chrom.end"</span>, nProf<span class="op">=</span><span class="fl">1203</span>, 
    uThresh<span class="op">=</span><span class="fl">0.2</span>, dThresh<span class="op">=</span><span class="fl">0.01</span><span class="op">)</span>

<span class="co">## Transfrom the mask segments into Genomic Ranges</span>
<span class="va">grAmpCNP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span>seqnames <span class="op">=</span> <span class="fu">Rle</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>, <span class="va">ampCNP</span><span class="op">$</span><span class="va">chrom</span><span class="op">)</span><span class="op">)</span>,
    ranges <span class="op">=</span> <span class="fu">IRanges</span><span class="op">(</span>start <span class="op">=</span> <span class="va">ampCNP</span><span class="op">$</span><span class="va">chrom.start</span>, end <span class="op">=</span> <span class="va">ampCNP</span><span class="op">$</span><span class="va">chrom.end</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span>,
    strand <span class="op">=</span> <span class="fu">Rle</span><span class="op">(</span><span class="fu">strand</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"*"</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">ampCNP</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co">## Create trellis graph</span>
<span class="fu"><a href="https://rdrr.io/pkg/gtrellis/man/gtrellis_layout.html">gtrellis_layout</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"hg19"</span>, category<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">7</span>,<span class="fl">9</span>,<span class="fl">15</span><span class="op">)</span><span class="op">)</span>, 
    n_track<span class="op">=</span><span class="fl">3</span>, nrow<span class="op">=</span><span class="fl">2</span>, track_height<span class="op">=</span><span class="fu">unit.c</span><span class="op">(</span><span class="fl">2</span><span class="op">*</span><span class="fu">grobHeight</span><span class="op">(</span><span class="fu">textGrob</span><span class="op">(</span><span class="st">"chr1"</span><span class="op">)</span><span class="op">)</span>, 
    <span class="fu">unit</span><span class="op">(</span><span class="fl">10</span>, <span class="st">"null"</span><span class="op">)</span>, <span class="fu">unit</span><span class="op">(</span><span class="fl">2</span>, <span class="st">"null"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co">## Add track for chromosome names</span>
<span class="fu"><a href="https://rdrr.io/pkg/gtrellis/man/add_track.html">add_track</a></span><span class="op">(</span>panel_fun <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">gr</span><span class="op">)</span> <span class="op">{</span>
    <span class="va">chr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/gtrellis/man/get_cell_meta_data.html">get_cell_meta_data</a></span><span class="op">(</span><span class="st">"name"</span><span class="op">)</span>  
    <span class="fu">grid.rect</span><span class="op">(</span>gp<span class="op">=</span><span class="fu">gpar</span><span class="op">(</span>fill<span class="op">=</span><span class="st">"#EEEEEE"</span><span class="op">)</span><span class="op">)</span>
    <span class="fu">grid.text</span><span class="op">(</span><span class="va">chr</span><span class="op">)</span> <span class="op">}</span><span class="op">)</span>

<span class="co">## Add segments representing the copy number events</span>
<span class="fu"><a href="https://rdrr.io/pkg/gtrellis/man/add_segments_track.html">add_segments_track</a></span><span class="op">(</span><span class="va">grAmpCNP</span>, value<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html">runif</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">grAmpCNP</span><span class="op">)</span><span class="op">)</span>, 
    gp<span class="op">=</span><span class="fu">gpar</span><span class="op">(</span>col<span class="op">=</span><span class="st">"blue"</span>, lwd<span class="op">=</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span>

<span class="co">## Add segments representing the masks</span>
<span class="va">maskFinal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">ampCNPmask</span><span class="op">)</span>
<span class="va">maskFinal</span><span class="op">$</span><span class="va">chrom</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">maskFinal</span><span class="op">$</span><span class="va">chrom</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/gtrellis/man/add_rect_track.html">add_rect_track</a></span><span class="op">(</span><span class="va">maskFinal</span>, h1<span class="op">=</span><span class="fl">0</span>, h2<span class="op">=</span><span class="fl">1</span>, gp<span class="op">=</span><span class="fu">gpar</span><span class="op">(</span>col<span class="op">=</span><span class="st">"red2"</span>, fill<span class="op">=</span><span class="st">"red2"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="CNprep_files/figure-html/makeMask-1.png" alt="Example of makeCNPmask() on a dataset of 1203 samples. The amplification events are shown, in blue, in the top section. Results of the makeCNPmask() are shown for four chromosomes, in red, in the bottom section. In this example, only chromosome 7 doesn't have any mask." width="700"><p class="caption">
Example of makeCNPmask() on a dataset of 1203 samples. The amplification events are shown, in blue, in the top section. Results of the makeCNPmask() are shown for four chromosomes, in red, in the bottom section. In this example, only chromosome 7 doesn’t have any mask.
</p>
</div>
</div>
<div id="apply-mask-to-a-table-of-copy-number-events" class="section level2">
<h2 class="hasAnchor">
<a href="#apply-mask-to-a-table-of-copy-number-events" class="anchor"></a>Apply mask to a table of copy number events</h2>
<p>The <em>applyCNPmask()</em> function … TODO</p>
</div>
</div>
<div id="using-parallelization" class="section level1">
<h1 class="hasAnchor">
<a href="#using-parallelization" class="anchor"></a>Using parallelization</h1>
<p>For experiments with many samples, one can take advantage of parallelized computation. Parallelizing <em>CNpreprocessing()</em> function can easily be accomplished by setting the <em>nJobs</em> parameter to a number higher than one. Beware that the parallelization is done per sample. So, the *nJobs should not be higher than the number of samples in the dataset.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Load needed datasets</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">segexample</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">ratexample</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">normsegs</span><span class="op">)</span>

<span class="co">## The number of threads is set to 4 (nJobs parameter)</span>
<span class="co">## Only the chromosomes 1 to 6 are analyzed (chromRange parameter)</span>
<span class="va">segtable</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CNpreprocessing.html">CNpreprocessing</a></span><span class="op">(</span>segall<span class="op">=</span><span class="va">segexample</span>, ratall<span class="op">=</span><span class="va">ratexample</span>, 
    idCol<span class="op">=</span><span class="st">"ID"</span>, startCol<span class="op">=</span><span class="st">"start"</span>, endCol<span class="op">=</span><span class="st">"end"</span>, chromCol<span class="op">=</span><span class="st">"chrom"</span>, 
    bpStartCol<span class="op">=</span><span class="st">"chrom.pos.start"</span>, bpEndCol<span class="op">=</span><span class="st">"chrom.pos.end"</span>, blsize<span class="op">=</span><span class="fl">50</span>,
    minJoin<span class="op">=</span><span class="fl">0.25</span>, cWeight<span class="op">=</span><span class="fl">0.4</span>, bsTimes<span class="op">=</span><span class="fl">50</span>, chromRange<span class="op">=</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>, nJobs<span class="op">=</span><span class="fl">4</span>,
    normalLength<span class="op">=</span><span class="va">normsegs</span><span class="op">[</span>,<span class="fl">1</span><span class="op">]</span>, normalMedian<span class="op">=</span><span class="va">normsegs</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">)</span></code></pre></div>
</div>
<div id="reproducible-research" class="section level1">
<h1 class="hasAnchor">
<a href="#reproducible-research" class="anchor"></a>Reproducible research</h1>
<p>To ensure reproducible results, <em>set.seed()</em> function should be call before <em>CNpreprocessing()</em>. Beware that the <em>nJobs</em> parameter must also be fixed. Change in the value of the <em>nJobs</em> parameter will lead to different results.</p>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<p>Here is the output of <em>sessionInfo()</em> on the system on which this document was compiled:</p>
<pre><code>## R version 4.0.2 (2020-06-22)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 10 x64 (build 19041)
## 
## Matrix products: default
## 
## locale:
## [1] LC_COLLATE=English_Canada.1252  LC_CTYPE=English_Canada.1252   
## [3] LC_MONETARY=English_Canada.1252 LC_NUMERIC=C                   
## [5] LC_TIME=English_Canada.1252    
## 
## attached base packages:
##  [1] grid      parallel  stats4    stats     graphics  grDevices utils    
##  [8] datasets  methods   base     
## 
## other attached packages:
##  [1] gtrellis_1.23.1      GenomicRanges_1.40.0 GenomeInfoDb_1.24.2 
##  [4] IRanges_2.22.2       CNprep_2.1.1         mclust_5.4.6        
##  [7] S4Vectors_0.26.1     BiocGenerics_0.34.0  BiocParallel_1.22.0 
## [10] BiocStyle_2.16.1    
## 
## loaded via a namespace (and not attached):
##  [1] XVector_0.28.0         compiler_4.0.2         BiocManager_1.30.10   
##  [4] highr_0.8              zlibbioc_1.34.0        bitops_1.0-6          
##  [7] tools_4.0.2            digest_0.6.27          evaluate_0.14         
## [10] memoise_2.0.0          rlang_0.4.10           yaml_2.2.1            
## [13] pkgdown_1.6.1.9000     xfun_0.21              fastmap_1.1.0         
## [16] GenomeInfoDbData_1.2.3 stringr_1.4.0          knitr_1.31            
## [19] GlobalOptions_0.1.2    desc_1.2.0             fs_1.5.0              
## [22] systemfonts_1.0.1      rprojroot_2.0.2        R6_2.5.0              
## [25] GetoptLong_1.0.2       textshaping_0.3.0      jpeg_0.1-8.1          
## [28] rmarkdown_2.6          bookdown_0.21          magrittr_2.0.1        
## [31] htmltools_0.5.1.1      assertthat_0.2.1       shape_1.4.4           
## [34] colorspace_2.0-0       circlize_0.4.11        ragg_1.1.0            
## [37] stringi_1.5.3          RCurl_1.98-1.2         cachem_1.0.4          
## [40] rjson_0.2.20           crayon_1.4.1</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-Carter">
<p>Carter, NP. 2007. “Methods and Strategies for Analyzing Copy Number Variation Using Dna Microarrays.” <em>Nat Genet</em> 39: S16–21. <a href="https://doi.org/10.1038/ng2028">https://doi.org/10.1038/ng2028</a>.</p>
</div>
<div id="ref-Duan">
<p>Duan, J, J-G Zhang, H-W Deng, and Y-P Wang. 2013. “Comparative Studies of Copy Number Variation Detection Methods for Next-Generation Sequencing Technologies.” <em>PLoS ONE</em> 8. <a href="https://doi.org/10.1371/journal.pone.0059128">https://doi.org/10.1371/journal.pone.0059128</a>.</p>
</div>
<div id="ref-Franch-Exposito2018a">
<p>Franch-Expósito, Sebastià, Clara Esteban-Jurado, Pilar Garre, Isabel Quintanilla, Saray Duran-Sanchon, Marcos Díaz-Gay, Laia Bonjoch, et al. 2018. “Rare germline copy number variants in colorectal cancer predisposition characterized by exome sequencing analysis.” <em>Journal of Genetics and Genomics</em> 45: 41–45. <a href="https://doi.org/10.1016/j.jgg.2017.12.001">https://doi.org/10.1016/j.jgg.2017.12.001</a>.</p>
</div>
<div id="ref-Kendall2011">
<p>Kendall, Jude, and Alexander Krasnitz. 2014. “Computational Methods for DNA Copy-Number Analysis of Tumors.” In <em>Cancer Genomics and Proteomics</em>, edited by Narendra Wajapeyee, 1176:243–59. Methods in Molecular Biology 6. New York, NY: Springer New York. <a href="https://doi.org/10.1007/978-1-4939-0992-6_20">https://doi.org/10.1007/978-1-4939-0992-6_20</a>.</p>
</div>
<div id="ref-Love">
<p>Love, Michael. 2019. <em>ExomeCopy: Copy Number Variant Detection from Exome Sequencing Read Depth</em>.</p>
</div>
<div id="ref-Pan2018">
<p>Pan, Xiaoyong, XiaoHua Hu, Yu-Hang Zhang, Lei Chen, LiuCun Zhu, ShiBao Wan, Tao Huang, and Yu-Dong Cai. 2018. “Identification of the copy number variant biomarkers for breast cancer subtypes.” <em>Molecular Genetics and Genomics</em> 294 (1): 95–110. <a href="https://doi.org/10.1007/s00438-018-1488-4">https://doi.org/10.1007/s00438-018-1488-4</a>.</p>
</div>
<div id="ref-Pelttari2018">
<p>Pelttari, L. M., H. Shimelis, H. Toiminen, A. Kvist, T. Törngren, Borg, C. Blomqvist, et al. 2018. “Gene-panel testing of breast and ovarian cancer patients identifies a recurrent RAD51C duplication.” <em>Clinical Genetics</em> 93 (3): 595–602. <a href="https://doi.org/10.1111/cge.13123">https://doi.org/10.1111/cge.13123</a>.</p>
</div>
<div id="ref-Seshan">
<p>Seshan, Venkatraman E., and Adam Olshen. 2019. <em>DNAcopy: DNA Copy Number Data Analysis</em>.</p>
</div>
<div id="ref-Talevich">
<p>Talevich, E, AH Shain, T Botton, and BC Bastian. 2016. “CNVkit: Genome-Wide Copy Number Detection and Visualization from Targeted Dna Sequencing.” <em>PLoS Comput Biol</em> 12 (4). <a href="https://doi.org/10.1371/journal.pcbi.1004873">https://doi.org/10.1371/journal.pcbi.1004873</a>.</p>
</div>
<div id="ref-Wang2018">
<p>Wang, Sheng, Jeffrey D. Mandelll, Yogesh Kumarr, Nawei Sunn, Montana T. Morris, Juan Arbelaez, Cara Nasello, et al. 2018. “De Novo Sequence and Copy Number Variants are Strongly Associated with Tourette Disorder and Implicate Cell Polarity in Pathogenesis.” <em>SSRN Electronic Journal</em> 24 (13): 3441–3454.e12. <a href="https://doi.org/10.2139/ssrn.3188485">https://doi.org/10.2139/ssrn.3188485</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Alex Krasnitz, Pascal Belleau, Astrid Deschênes, Guoli Sun.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
